### PR TITLE
Caching failed requests, support blocked URL patterns

### PR DESCRIPTION
**Pull request by JasonPapanicholasWF**
## Description

- When requests fail, cache them anyway
- More logging
- Ability to pass in blocked URL patterns

tested: true

### CODE REVIEW

File: packages/proxy/src/cache-helpers.ts [hunk 1]
```diff
@@ -42,15 +42,26 @@ export const calculateCacheKey = (
     url: request.url,
   };
 
-  // for now we only differentiate between graphql and non graphql requests, but we can improve
-  if (request.url?.startsWith('/graphql')) {
-    key = `/graphql?hash=${md5(requestBody.toString())}`;
+  if (request.method === 'POST') {
+    key = `${request?.url?.split(/[?#]/)[0]}?hash=${md5(
```
Comment (roippi) [inline packages/proxy/src/cache-helpers.ts]: will this still work if the original URL also has query params?  (i.e. already has a `?`)

File: packages/proxy/src/proxy-worker.ts [hunk 2]
```diff
@@ -36,31 +41,33 @@ export const handleProxyResponse = async ({
...
-    console.error(`Cannot process response to ${originalRequest?.url}`);
   }
 };
 
 export const buildProxyWorker = ({cache}: {cache: ProxyCache}) => {
   const proxy = createProxyServer({
     selfHandleResponse: true,
+    autoRewrite: true,
+    changeOrigin: true,
+    followRedirects: true,
```
Comment (roippi) [inline packages/proxy/src/proxy-worker.ts:65]: might eventually want to provide config for settings like this one.  just making a note of it :)

File: packages/proxy/src/cache-helpers.ts [hunk 3]
```diff
@@ -42,15 +42,26 @@ export const calculateCacheKey = (
     url: request.url,
   };
 
-  // for now we only differentiate between graphql and non graphql requests, but we can improve
-  if (request.url?.startsWith('/graphql')) {
-    key = `/graphql?hash=${md5(requestBody.toString())}`;
+  if (request.method === 'POST') {
```
Comment (spember) [inline packages/proxy/src/cache-helpers.ts]: this should probably include not just post, but any request that has a requestBody at all (e.g. a GET could sneak one in, as would a PUT)

**Review by roippi — APPROVED** [review]

File: packages/proxy/src/reverse-proxy.ts [hunk 4]
```diff
@@ -26,22 +27,22 @@ export const handleIncomingRequest = async ({
...
 
     response.writeHead(cachedResponse.status, cachedResponse.headers);
     response.write(cachedResponse.data);
     response.end();
   } else {
-    console.log(`🆕 Key created - ${cacheKey.slice(0, 150)}`);
-
+    console.log(
+      `✅ Key created - ${trimKey(cacheKey)}`,
+      request.method === 'POST' && `\nPOST Data - ${requestData.toString()}`
```
Comment (spember) [inline packages/proxy/src/reverse-proxy.ts]: again, I worry about only using data for POST

**Review by spember — COMMENTED** [review]
One suggestion on expanding the scope of using requestBody as part of the cache key

File: packages/proxy/src/cache-helpers.ts (hunk 1 above)
Comment (JasonPapanicholasWF) [reply packages/proxy/src/cache-helpers.ts]: I can test that again

Comment (JasonPapanicholasWF) [reply packages/proxy/src/cache-helpers.ts]: Or better, I can write a test

**Pull request merged**